### PR TITLE
4pu multi

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

File: src/analog/commands/downlink/getChannelsStatus.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+export const name: types.TCommandName = 'getChannelsStatus';
+export const headerSize = 3;
+
+
+export const examples: command.TCommandExamples = {
+    'request the status of all channels': {
+        id,
+        name,
+        headerSize,
+        parameters: null,
```
Comment (DarkPark) [inline src/analog/commands/downlink/getChannelsStatus.ts:37]: we usually use `parameters: {}`

File: src/analog/commands/uplink/getChannelsMap.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/**
+ * Information about the current role of channels on the device,
```
Comment (DarkPark) [inline src/analog/commands/uplink/getChannelsMap.ts:2]:
```suggestion
 * Information about the current role of channels on the device.
```

File: src/analog/commands/uplink/getChannelsStatus.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
+/**
+ * Information about the current status of channels on the device,
```
Comment (DarkPark) [inline src/analog/commands/uplink/getChannelsStatus.ts:2]:
```suggestion
 * Information about the current status of channels on the device.
```

File: src/analog/commands/uplink/getChannelsStatus.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+import * as command from '../../utils/command.js';
+import * as channelsTypes from '../../constants/channelsTypes.js';
+
+
+export const id: types.TCommandId = 0x321f;
+export const name: types.TCommandName = 'getChannelsStatus';
+export const headerSize = 3;
+
+
+interface IBinarySensorStatus {
```
Comment (DarkPark) [inline src/analog/commands/uplink/getChannelsStatus.ts:40]: it's better to put interfaces before constants

File: src/analog/commands/uplink/getChannelsStatus.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+/**
+ * Decode command parameters.
+ *
+ * @param data - only body (without header)
+ * @returns command payload
+ */
+export const fromBytes = ( data: types.TBytes ): Array<IChannelStatus> => {
+    const buffer: IBinaryBuffer = new BinaryBuffer(data);
+    const result: Array<IChannelStatus> = [];
+
```
Comment (DarkPark) [inline src/analog/commands/uplink/getChannelsStatus.ts:163]:
```suggestion
```

File: src/analog/utils/CommandBinaryBuffer.ts [hunk 6]
```diff
@@ -616,6 +620,45 @@ interface IParameterNbiotSim {
     pin: types.TUint16
 }
 
+/**
+ * Idle channel
+ */
+interface IParameterIdleChannel {
```
Comment (DarkPark) [inline src/analog/utils/CommandBinaryBuffer.ts:626]:
```suggestion
interface IParameterIdleChannel {}
```
?

File: src/analog/utils/CommandBinaryBuffer.ts [hunk 7]
```diff
@@ -616,6 +620,45 @@ interface IParameterNbiotSim {
...
+}
+
+/**
+ * Power channel settings
+ */
+interface IParameterPowerChannel {
```
Comment (DarkPark) [inline src/analog/utils/CommandBinaryBuffer.ts:632]: is it ok to be empty?

File: src/analog/utils/CommandBinaryBuffer.ts [hunk 8]
```diff
@@ -2394,4 +2500,84 @@ CommandBinaryBuffer.prototype.setDataSegment = function ( parameters: IDataSegme
...
+    this.setUint8(parameters.highTemperatureThreshold);
+    this.setUint8(parameters.lowTemperatureThreshold);
+};
+
+
+CommandBinaryBuffer.prototype.getChannelType = function (): IParameterChannelType {
+    const channel = getChannelValue(this);
+    const type = this.getUint8();
+    let parameters = {};
+
```
Comment (DarkPark) [inline src/analog/utils/CommandBinaryBuffer.ts:2541]:
```suggestion
```

File: src/analog/commands/downlink/getChannelsMap.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Command to get about the current role of channels on the device,
```
[the remark this answers is not in the document]
Comment (Pnom) [reply src/analog/commands/downlink/getChannelsMap.ts]: I renamed the command from getChannelsMap to getChannelsTypes and updated the comments

File: src/analog/commands/downlink/getChannelsStatus.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Command to get the status of device channels.
+ * Optionally, specify a channels to get the status of the subsystem assigned to that channels.
+ * Without a channels , the command returns the status of all channels.
```
[the remark this answers is not in the document]
Comment (Pnom) [reply src/analog/commands/downlink/getChannelsStatus.ts]: fixed

File: src/analog/commands/downlink/getChannelsStatus.ts [hunk 11]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Command to get the status of device channels.
+ * Optionally, specify a channels to get the status of the subsystem assigned to that channels.
```
[the remark this answers is not in the document]
Comment (Pnom) [reply src/analog/commands/downlink/getChannelsStatus.ts]: fixed

File: src/analog/commands/downlink/getChannelsStatus.ts (hunk 1 above)
Comment (Pnom) [reply src/analog/commands/downlink/getChannelsStatus.ts]: fixed

File: src/analog/commands/uplink/getChannelsMap.ts (hunk 2 above)
Comment (Pnom) [reply src/analog/commands/uplink/getChannelsMap.ts]: fixed

File: src/analog/commands/uplink/getChannelsMap.ts [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+export const headerSize = 3;
+
+
+interface IChannelsMap {
+    channelsMap: Array<types.TUint8>
+}
+
+
+export const examples: command.TCommandExamples = {
+    'channels map. [POWER_CHANNEL (2), BINARY_SENSOR (3), TEMPERATURE_SENSOR (4), IDLE (0)]': {
```
[the remark this answers is not in the document]
Comment (Pnom) [reply src/analog/commands/uplink/getChannelsMap.ts]: fixed

File: src/analog/commands/uplink/getChannelsStatus.ts (hunk 4 above)
Comment (Pnom) [reply src/analog/commands/uplink/getChannelsStatus.ts:35]: fixed

File: src/analog/commands/uplink/getChannelsStatus.ts [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+ *
+ * // decoded payload
+ * const parameters = getChannelsStatus.fromBytes(bytes);
+ *
+ * console.log(parameters);
+ * // output:
+ * [{
+ *     type: 2,
+ *     status: {
+ *         channel: 0,
```
[the remark this answers is not in the document]
Comment (Pnom) [reply src/analog/commands/uplink/getChannelsStatus.ts]: fixed

File: src/analog/commands/uplink/getChannelsStatus.ts (hunk 5 above)
Comment (Pnom) [reply src/analog/commands/uplink/getChannelsStatus.ts:163]: fixed

File: src/analog/utils/CommandBinaryBuffer.ts (hunk 8 above)
Comment (Pnom) [reply src/analog/utils/CommandBinaryBuffer.ts:2539]: fixed

File: src/analog/utils/CommandBinaryBuffer.ts (hunk 7 above)
Comment (Pnom) [reply src/analog/utils/CommandBinaryBuffer.ts]: I added the IParameterEmpty interface as an analog to the IEmptyCommandParameters interface and used it to replace the empty IParameterIdleChannel and IParameterPowerChannel interfaces.

File: src/analog/utils/CommandBinaryBuffer.ts (hunk 6 above)
Comment (Pnom) [reply src/analog/utils/CommandBinaryBuffer.ts]: I added the IParameterEmpty interface as an analog to the IEmptyCommandParameters interface and used it to replace the empty IParameterIdleChannel and IParameterPowerChannel interfaces.

File: src/analog/commands/downlink/getChannelsTypes.ts [hunk 14]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Command to get the types of channels on the device.
+ * like IDLE, TEMPERATURE_SENSOR, or BINARY_SENSOR.
```
Comment (DarkPark) [inline src/analog/commands/downlink/getChannelsTypes.ts:3]: is it a new sentence?

Comment (Pnom) [reply src/analog/commands/downlink/getChannelsTypes.ts:3]: fixed

File: src/analog/commands/uplink/getChannelsStatus.ts (hunk 3 above)
Comment (Pnom) [reply src/analog/commands/uplink/getChannelsStatus.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**